pkg/utils/whisper/whisper_test.go

```diff
@@ -0,0 +1,56 @@
+package whisper_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	api "github.com/rotationalio/whisper/pkg/api/v1"
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/utils/whisper"
+)
+
+func TestCreateWhisperLink(t *testing.T) {
+	// CreateWhisperLink should throw an error when passed an empty secret
+	accesses := 3
+	oneWeek := time.Now().AddDate(0, 0, 7)
+	link, err := whisper.CreateWhisperLink("", "not empty", accesses, oneWeek)
+	require.Equal(t, link, "")
+	require.EqualError(t, err, "a secret is required to generate a Whisper link")
+
+	// CreateWhisperLink should throw an error when passed an empty password
+	link, err = whisper.CreateWhisperLink("not empty", "", accesses, oneWeek)
+	require.Equal(t, link, "")
+	require.EqualError(t, err, "a password is required to generate a Whisper link")
+
+	// CreateWhisperLink should throw an error when passed an expiration date in the past
+	yesterday := time.Now().Add(-time.Hour * 24)
+	link, err = whisper.CreateWhisperLink("not empty", "not empty", accesses, yesterday)
+	require.Equal(t, link, "")
+	require.EqualError(t, err, "the expiration date for the secret must be in the future")
+
+	// Pass in valid arguments and check that the returned URL is valid
+	link, err = whisper.CreateWhisperLink("this is a secret", "password", accesses, oneWeek)
```

Comment:
This creates a live secret - which is a pretty good test to have; though I would prefer our tests not be dependent on Whisper as an externality. May I suggest that we guard the live test with an environment variable; e.g. to run the live test we have to specify:

```
$ GDS_TEST_LIVE_WHISPER=1 go test ./...
```

And that for standard testing we create an `httptest.NewTestServer` and inject the client to make sure the request is what we'd like? Feel free to talk to Patrick about this methodology.